swarm/network/hive.go

```diff
@@ -192,6 +199,7 @@ func (self *Hive) Start(id discover.NodeID, listenAddr func() string, connectPee
 func (self *Hive) keepAlive() {
 	alarm := time.NewTicker(time.Duration(self.callInterval)).C
 	for {
+		peersNumGauge.Update(int64(self.kad.Count()))
```

Comment:
Should this not only be updated when the peer count changes?

Reply:
I have thought of the same thing. The reason I resorted to including it here was that 
1. We then need to also do it on a remove
2. The time frame for add / remove is quite different and not uniform
But I have no strong opinion on this, it could be removed from the loop and updated on add/remove (@nonsense, what happens if the network is stable, and not much peers are added/removed - if the Gauge is being used for number of peers, does that somehow affect the stats if the value is old?)

Reply:
@holisticode Gauge always has a stable value - if you don't update it, the last used value is emitted. If you update it multiple times during a flush period, only the last value will be sent to the backend.